internals-js/src/definitions.ts

```diff
@@ -1054,24 +920,76 @@ export class CoreFeatures {
     if (directive.definition?.name !== this.coreItself.nameInSchema) {
       return undefined;
     }
-    const args = (directive as Directive<SchemaDefinition, CoreDirectiveArgs>).arguments();
-    const url = FeatureUrl.parse(args.feature);
+    const typedDirective = directive as Directive<SchemaDefinition, CoreOrLinkDirectiveArgs>
+    const args = typedDirective.arguments();
+    const url = this.coreDefinition.extractFeatureUrl(args);
     const existing = this.byIdentity.get(url.identity);
     if (existing) {
       throw error(`Duplicate inclusion of feature ${url.identity}`);
     }
-    const feature = new CoreFeature(url, args.as ?? url.name, directive, args.for);
+    const imports = extractCoreFeatureImports(typedDirective);
+    const feature = new CoreFeature(url, args.as ?? url.name, directive, imports, args.for);
     this.add(feature);
+    directive.schema().blueprint.onAddedCoreFeature(directive.schema(), feature);
     return feature;
   }
 
   private add(feature: CoreFeature) {
     this.byAlias.set(feature.nameInSchema, feature);
     this.byIdentity.set(feature.url.identity, feature);
   }
+
+  sourceFeature(element: DirectiveDefinition | NamedType): CoreFeature | undefined {
+    const isDirective = element instanceof DirectiveDefinition;
+    const splitted = element.name.split('__');
+    if (splitted.length > 1) {
+      return this.byAlias.get(splitted[0]);
```

Comment:
If it did, this would throw if a user decided to use `__` within a name but without meaning to use anything core schema related, and since `__` in the middle of names is not invalid graphQL, I don't know if we want that (note that I wouldn't mind terribly making `__` in names "reserved", but I'm not particular in favor either because I don't think we need such restriction).